code/modules/recycling/disposal-unit.dm

```diff
@@ -527,6 +527,160 @@
+	icon = 'icons/obj/atmospherics/pipes/trapdoor.dmi'
+	icon_state = "closed"
+	mode = 0
+	layer = 2.01
+	var/id = 1
+	var/auto_close = 1200
+	var/auto_close_on_mob = 100
+	var/sound_open = 'sound/machines/blast_door.ogg'
+	var/sound_close = 'sound/machines/blast_door.ogg'
+	var/open = FALSE
+
+/obj/machinery/disposal/trapdoor/proc/open()
+	if(flushing)
+		return
+	if(open)
+		return
+	flick("opening", src)
+	playsound(src.loc, sound_open, 100, 1)
+	icon_state = "open"
+	spawn(5)
+	for(var/mob/living/M in loc)
+		if(!M.floating)
+			M.loc = src
+			trap_flush()
+			if(auto_close_on_mob)
+				spawn(auto_close_on_mob)
+					close()
+	for(var/obj/item/O in loc)
+		if(!O.throwing || !O.anchored)
+			O.loc = src
+			trap_flush()
+	open = TRUE
+	if(auto_close)
+		spawn(auto_close)
+			close()
+	return 1
+
+
+/obj/machinery/disposal/trapdoor/proc/close()
+	if(open)
+		flick("closing", src)
+		icon_state = "closed"
+		playsound(src.loc, sound_open, 100, 1)
+		open = FALSE
+		return 1
+
+/obj/machinery/disposal/trapdoor/New(loc,var/obj/structure/disposalconstruct/make_from)
+	..()
+	stored.ptype = DISP_END_CHUTE
+	spawn(5)
+		trunk = locate() in loc
+		if(trunk)
+			trunk.linked = src
+
+/obj/machinery/disposal/trapdoor/Crossed(AM as mob|obj)
+	if(open)
+		if(istype(AM, /mob/living/))
+			var/mob/living/M = AM
+			if(M.floating)
+				return
+			M.loc = src
+			trap_flush()
+			if(auto_close_on_mob)
+				spawn(auto_close_on_mob)
+					close()
+			return
+		if(istype(AM, /obj/item))
+			var/obj/item/O = AM
+			spawn(5)
+				if(O.throwing || O.anchored)
+					return
+				else if(O.loc == src.loc)
+					O.loc = src
+					trap_flush()
+
+/obj/machinery/disposal/trapdoor/MouseDrop_T(mob/living/target, mob/living/user)
+	if (!open)
+		return
+	if(istype(target) && user == target)
+		push_mob_in(target, user)
+		return 1
+
+/obj/machinery/disposal/trapdoor/proc/push_mob_in(mob/living/target, mob/living/user)
+	if(target.buckled)
+		return
+	add_fingerprint(user)
+	if(user == target)
+		if(target.floating)
+			user.visible_message("[user] is attempting to dive into [src].", \
+				"<span class='notice'>You start diving into [src]...</span>")
+			if(do_mob(target, user, 10))
+				if (!loc)
+					return
+				target.loc = src
+				user.visible_message("[user] dives into [src].", \
+					"<span class='notice'>You dive into [src].</span>")
+				update()
+				sleep(5)
+				trap_flush()
+				if(auto_close_on_mob)
+					spawn(auto_close_on_mob)
+						close()
+		else
+			user.visible_message("[user] is attempting to step on the edge of [src].", \
+				"<span class='notice'>You start attempting to step on the edge of [src]...</span>")
+			if(do_mob(target, user, 30))		//Add 25% chance of failing "You slip and fall in to [src], 75% if clumsy "You step.. nope too funny you backwards flip in to [src], HONK!", 75% if braindamaged "You close your eyes so [src] cant detect you, and boldly step forward.", 100% if wet floor (noslip/galosh check) "You slip, and fall in to [src]."
```

Comment:
//Add 25% chance of failing "You slip and fall in to [src], 75% if clumsy "You step.. nope too funny you backwards flip in to [src], HONK!", 75% if braindamaged "You close your eyes so [src] cant detect you, and boldly step forward.", 100% if wet floor (noslip/galosh check) "You slip, and fall in to [src]."

after the if !loc return but before setting loc:
var/chance = 25 // normal chance, 25% to fall inside
var/turf/T = get_turf(src) // to check if turf's wet
var/message/U = "fall inside"
var/message/U = "falls inside"
if(user.disabilities & CLUMSY)
     chance = 80
     M = "accidentally do a backward flip, falling inside"
     U = "accidentally does a backward flip, falling inside"
else if(user.getBrainLoss() >= 50)
     chance = 70
     M = "close your eyes and boldly step forward"
     U = "closes his eyes and boldly steps forward"
else if(T.wet && isobj(user.shoes) &&  user.shoes.flags&NOSLIP)
     chance = 60
     M = "slip and fall inside"
     U = "slips and falls inside"
if(prob(chance))
     user.visible_message("<span class='userdanger'>[U] \the [src]!</span>", <span class='danger'>You [M] \the [src]!)
     user.loc = loc
     user.Stun(10)

this was coded rapidly so you should check if it works as wanted aye